src/interfaces.js

```diff
@@ -73,5 +73,65 @@ interface RouteConfig {
   */
   caseSensitive?: boolean;
 
+  /**
+  * Add to specify an activation strategy if it is always the same and you do not want that
+  * to be in your view-model code. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  activationStrategy?: string;
+  
   [x: string]: any;
 }
+
+type ShouldContinue = boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+type Potential = Promise<void>|PromiseLike<void>|IObservable;
```

Comment:
> I'll wait for any other input

@EisenbergEffect, could someone from the Aurelia team review this PR as well? I don't have any more suggestions :)

> I'll squash the commits or I shouldn't?

Yeah, I guess You'll be asked to do so